File: src/main/java/com/hea3ven/dulcedeleche/ModDulceDeLeche.java
package com.hea3ven.dulcedeleche;

import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.common.Mod.EventHandler;
import net.minecraftforge.fml.common.event.FMLInitializationEvent;
import net.minecraftforge.fml.common.event.FMLPostInitializationEvent;
import net.minecraftforge.fml.common.event.FMLPreInitializationEvent;

import com.hea3ven.tools.bootstrap.Bootstrap;

@Mod(modid = ModDulceDeLeche.MODID, version = ModDulceDeLeche.VERSION,
		dependencies = "required-after:Forge@[11.14.3.1513,)")
public class ModDulceDeLeche {

	public static final String MODID = "dulcedeleche";
	public static final String VERSION = "1.0.0";

	private final ProxyCommonDulceDeLeche proxy = new ProxyCommonDulceDeLeche();

	static {
		Bootstrap.init();
	}

	@EventHandler
	public void onPreInitEvent(FMLPreInitializationEvent event) {
		proxy.onPreInitEvent(event);
	}

	@EventHandler
	public void onInitEvent(FMLInitializationEvent event) {
		proxy.onInitEvent(event);
	}

	@EventHandler
	public void onPostInitEvent(FMLPostInitializationEvent event) {
		proxy.onPostInitEvent(event);
	}
}

File: src/main/java/com/hea3ven/dulcedeleche/ModDulceDeLeche.kt
package com.hea3ven.dulcedeleche

import com.hea3ven.tools.bootstrap.Bootstrap
import net.minecraftforge.fml.common.Mod
import net.minecraftforge.fml.common.event.FMLInitializationEvent
import net.minecraftforge.fml.common.event.FMLPostInitializationEvent
import net.minecraftforge.fml.common.event.FMLPreInitializationEvent

@Mod(modid = ModDulceDeLeche.MODID, version = ModDulceDeLeche.VERSION,
		dependencies = ModDulceDeLeche.DEPENDENCIES,
		guiFactory = "com.hea3ven.dulcedeleche.DulceDeLecheModGuiFactory",
		updateJSON = "https://raw.githubusercontent.com/hea3ven/DulceDeLeche/version/media/update.json")
class ModDulceDeLeche {
	@Mod.EventHandler
	fun OnPreInit(event: FMLPreInitializationEvent?) {
		proxy.onPreInitEvent(event)
	}

	@Mod.EventHandler
	fun OnInit(event: FMLInitializationEvent?) {
		proxy.onInitEvent(event)
	}

	@Mod.EventHandler
	fun OnPostInit(event: FMLPostInitializationEvent?) {
		proxy.onPostInitEvent(event)
	}

	companion object {
		const val MODID = "dulcedeleche"
		const val VERSION = "1.9.4-1.0.1"
		const val DEPENDENCIES = "required-after:Forge@[12.17.0.1954,)"

		const val guiIdAssembler = 0

		init {
			Bootstrap.init()
		}

		val proxy = ProxyModDulceDeLeche()
	}
}

